tools/reactive-controllers/src/downstate.ts

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import type { LitElement, ReactiveController } from 'lit';
+
+export class DownState implements ReactiveController {
+    private host: LitElement;
+    private abortController: AbortController;
+
+    constructor(host: LitElement) {
+        this.host = host;
+        this.abortController = new AbortController();
+        this.host.addController(this);
+    }
+
+    public hostConnected(): void {
+        this.manage();
+    }
+
+    public hostDisconnected(): void {
+        this.removeEventListeners();
+        this.abortController.abort();
+    }
+
+    public manage(): void {
+        this.addEventListeners();
+    }
+
+    public unmanage(): void {
+        this.removeEventListeners();
+    }
+
+    addEventListeners(): void {
+        this.host?.addEventListener('pointerdown', this.handlePointerDown, {
+            signal: this.abortController.signal,
+        });
+        this.host?.addEventListener('pointerup', this.handlePointerUp, {
+            signal: this.abortController.signal,
+        });
+    }
+
+    removeEventListeners(): void {
+        this.host?.removeEventListener('pointerdown', this.handlePointerDown);
+        this.host?.removeEventListener('pointerup', this.handlePointerUp);
+    }
+
+    public handlePointerDown = (): void => {
+        this.updateDownStateStyles(true);
+    };
+
+    public handlePointerUp = (): void => {
+        this.updateDownStateStyles(false);
+    };
```

Comment:
This applies these changes synchronously, which is probably correct. However, the majority of our render reactions are async, due to the reactivity of LitElement. Do you se any issue with this?

Reply:
I feel updating the down state styles is a UI-specific operation and not directly related to LitElement's reactivity, handling it synchronously shouldn't cause any problems, but would love to hear your thoughts why you foresee some issues here?

Reply:
This has been updated!